airflow/executors/fargate_executor.py

```diff
@@ -0,0 +1,336 @@
+                               f'skipping for now on strike #{task.api_failure_count}')
+
+    def execute_async(self, key: TaskInstanceKeyType, command: CommandType,
+                      queue=None, executor_config=None):
+        """
+        This method will execute the command asynchronously.
+        """
+        # run a task
+        task_execution_api = self.executor_config_function(command)
+        try:
+            task_response = self.ecs.run_task(**task_execution_api)
+            self.log.debug(f'Executor responded with "{task_response}"')
+            if task_response['failures'] or not task_response['tasks']:
+                raise FargateError(', '.join([str(f) for f in task_response['failures']]))
+            task = FargateTask(task_response['tasks'][0])
+            self.log.info(f'Executor running task "{key}" on "{task.task_arn}"')
+            self.active_workers.add_task(task, key)
+        except (FargateError, Boto3Error) as e:
+            self.log.error(f'Executor failed to run task {key} with the following exception: {str(e)}')
+            self.fail(key)
+
+    def end(self, heartbeat_interval=10):
+        """
+        This method is called when the caller is done submitting job and is
+        wants to wait synchronously for the job submitted previously to be
+        all done.
+        """
+        while True:
+            self.sync()
+            if not len(self.active_workers):
+                break
+            time.sleep(heartbeat_interval)
+
+    def terminate(self):
+        """
+        This method is called when the daemon receives a SIGTERM
+        """
+        for arn in self.active_workers.get_all_arns():
+            self.ecs.stop_task(
+                cluster=self.cluster,
+                task=arn,
+                reason='Airflow Executor received a sig-term'
+            )
+        self.end()
+
+
+def default_task_id_to_fargate_options_function():
+    """
+    This is a function which returns a function. The outer function takes no arguments, and returns the inner function.
+    The inner function takes in an airflow CLI command an outputs a json compatible with the boto3 run_task API
+    linked above. In other words, if you don't like the way I call the boto3 Fargate API then call it yourself by
+    overriding the airflow config file.
+
+    i.e: execution_config_function = airflow.executors.fargate_executor.default_task_id_to_fargate_options_function
+    """
+    # Absolutely mandatory configurations
+    cluster = configuration.conf.get('fargate', 'cluster')
+
+    # grab a few variables
+    task_definition = configuration.conf.get('fargate', 'task_definition')
+    container_name = configuration.conf.get('fargate', 'container_name')
+    security_groups = configuration.conf.get('fargate', 'security_groups').split(',')
+
+    launch_type = 'FARGATE'
+    if configuration.conf.has_option('fargate', 'launch_type'):
+        launch_type = configuration.conf.get('fargate', 'launch_type')
```

Comment:
```suggestion
        launch_type = configuration.conf.get('fargate', 'launch_type', fallback='FARGATE')
```
Instead of checking for option you can use fallback for `conf.get`, I think it will simplify things a little bit.

Reply:
Oh, today I learned

Reply:
Yeah, this's not obvious when one look at `get` definition